server/src/main/java/org/elasticsearch/index/mapper/ObjectMapper.java

```diff
@@ -565,6 +576,30 @@ private static Map<String, Mapper> buildMergedMappers(
             }
             return Map.copyOf(mergedMappers);
         }
+
+        private static void addNewObjectMapper(
+            MergeReason reason,
+            MapperMergeContext objectMergeContext,
+            Map<String, Mapper> mergedMappers,
+            ObjectMapper objectMapper
+        ) {
+            // first try to add the full object, with all it's sub-fields
+            boolean fullObjectAdded = objectMergeContext.addFieldIfPossible(objectMapper, o -> mergedMappers.put(o.simpleName(), o));
+            if (fullObjectAdded) {
+                return;
+            }
+            // there's not enough capacity for the whole object mapper,
+            // so we're just trying to add the shallow object, without it's sub-fields
+            ObjectMapper shallowObjectMapper = objectMapper.withoutMappers();
```

Comment:
is there a test that fails if this fallback is entirely removed? Do we verify that it does what it's supposed to do? I am thinking that despite the index setting is not yet wired, we can add unit tests to ObjectMapperMergeTests with an real budget and leverage this codepath?

Reply:
Yes, I've already added `org.elasticsearch.index.mapper.ObjectMapperMergeTests#testMergeWithLimitObjectField` to test for this case.